### PR TITLE
SassService: use sassphp extension for rendering SASS

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/PLATFORM-1313

Use libsass-powered PHP extension instead of external Ruby script.

Additionally change how `AssetsManager` builds memcache keys to support per-environment prefixes (avoid cache polution between production and staging / sandboxes)
### Performance comparison

Rendering of `/skins/oasis/css/oasis.scss` (the main CSS of Oasis skin):
- with `sassphp` extension (`LibSassCompiler` class) - ~0,48 s
- external Ruby script (`ExternalRubyCompiler` class) - 6 sec (~12 times slower)

**Do not merge it yet** - we need to deploy PHP extension first.

@michalroszka / @wladekb / @RafLeszczynski / @owend

### CODE REVIEW

File: includes/wikia/services/sass/Compiler/LibSassCompiler.class.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	\$sassVariables: $sassVariables;
+
+	@if map-has-key(\$sassVariables, \$varName) {
+		@return map-get(\$sassVariables, \$varName);
+	}
+	@else {
+		@return \$defaultValue;
+	}
+}
+SASS;
```
Comment (macbre) [inline includes/wikia/services/sass/Compiler/LibSassCompiler.class.php:81]: `/extensions/wikia/SASS/wikia_sass.rb` ported to SASS function. This one allows us to inject user-defined variables into SASS stylesheets.

File: includes/wikia/services/sass/SassService.class.php [hunk 2]
```diff
@@ -23,9 +24,9 @@
  * @author Władysław Bodzek <wladek@wikia-inc.com>
  *
  */
-class SassService extends WikiaObject {
```
Comment (wladekb) [inline includes/wikia/services/sass/SassService.class.php:26]: why not extend WikiaObject?

Comment (macbre) [inline includes/wikia/services/sass/SassService.class.php:26]: I want to have a private constructor here (docs say that static `newFrom...` functions should be used instead).

**Comment by macbre** [comment]
Tests are failing as we're waiting for https://wikia-inc.atlassian.net/browse/OPS-6839

```
Running 'SassServiceTest'...F
Fatal error: Class 'Sass' not found in /var/lib/jenkins/workspace/unitTests_php_branches/includes/wikia/services/sass/Compiler/LibSassCompiler.class.php on line 30
```

**Comment by macbre** [comment]
Jenkins now passes - http://qa-s2.wikia-prod:8080/job/unitTests_php_branches/18577/console

```
Running 'SassServiceTest'...........done in 29.04 ms [0 slow tests]
```

**Pull request merged**